kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js

```diff
@@ -37,13 +39,55 @@ grCollectionsPanel.controller('GrCollectionsPanelCtrl', [
     ctrl.selectedCollections = selectedCollections;
 }]);
 
+grCollectionsPanel.factory('collectionsTreeState', ['$window', function($window) {
+    // TODO: Add garbage collection to state.
+    const localStorageKey = 'collectionsTreeOpen';
+    const jsonStr = $window.localStorage.getItem(localStorageKey) || '[]';
+
+    // A little bit of superstition in case this was set weirdly before.
+    let jsonArr = [];
+    try {
+        let jsonArr = JSON.parse(jsonStr);
+        if (!Array.isArray(jsonArr)) {
+            throw new Error(`Invalid ${localStorageKey} json: ${jsonArray}`);
+        }
```

Comment:
The `let` in `let jsonArr = JSON.parse(jsonStr);` is block-scoped (i.e. scoped to the `try`), so it locally shadows the parent `let jsonArr = [];`, which is never updated and always equal to `[]`.

[Reduced example](https://google.github.io/traceur-compiler/demo/repl.html#var%20json%20%3D%20'%7B%7D'%0A%0Afunction%20parse(json)%20%7B%0A%20%20let%20jsonArr%20%3D%20%5B%5D%3B%0A%20%20try%20%7B%0A%20%20%20%20let%20jsonArr%20%3D%20JSON.parse(json)%3B%0A%20%20%20%20if%20(!Array.isArray(jsonArr))%20%7B%0A%20%20%20%20%20%20throw%20new%20Error(%60Invalid%20%24%7BlocalStorageKey%7D%20json%3A%20%24%7BjsonArray%7D%60)%3B%0A%20%20%20%20%7D%0A%20%20%7D%20catch%20(e)%20%7B%0A%20%20%20%20console.log(%22ERROR%3A%22%2C%20e)%0A%20%20%7D%0A%20%20return%20jsonArr%3B%0A%7D%0A%0Aconsole.log(%22parsing%20%22%2C%20'%7B%7D'%2C%20%22expected%3A%20%5B%5D%2C%20actual%3A%20%22%2C%20parse('%7B%7D'))%0Aconsole.log(%22parsing%20%22%2C%20'%5B1%2C2%5D'%2C%20%22expected%3A%20%5B1%2C2%5D%2C%20actual%3A%20%22%2C%20parse('%5B1%2C2%5D'))%0Aconsole.log(%22parsing%20%22%2C%20'%7B'%2C%20%22expected%3A%20%5B%5D%2C%20actual%3A%20%22%2C%20parse('%7B'))) (open the console)

Removing the inner `let` fixes the issue, but breaks the case if the parsed JSON isn't an Array.